packages/wallets/src/features/accounts/modules/TaxInformation/hooks/useTaxInformation.ts

```diff
@@ -12,13 +13,32 @@ type TTaxInformationValues = {
     taxResidence: THooks.AccountSettings['tax_residence'];
 };
 
-const useTaxInformation = () => {
+const useTaxInformation = (selectedJurisdiction: string) => {
     const { localize } = useTranslations();
-    const { data: residenceList, isLoading, isSuccess: isResidenceListSuccess } = useResidenceList();
+    const {
+        data: residenceList,
+        isLoading: isResidenceListLoading,
+        isSuccess: isResidenceListSuccess,
+    } = useResidenceList();
     const { data: accountSettings, error, isSuccess: isAccountSettingsSuccess, update } = useSettings();
+    const { data: landingCompanyDetailsData, isLoading: isLandingCompanyDetailsLoading } = useQuery(
+        'landing_company_details',
+        {
+            options: {
+                enabled: isAccountSettingsSuccess,
+            },
+            payload: {
+                country: accountSettings.country_code ?? '',
+                landing_company_details:
+                    selectedJurisdiction as TSocketRequestPayload<'landing_company_details'>['payload']['landing_company_details'],
```

Comment:
@aum-deriv 
maybe we can proper define selectedJurisdiction type in the root (in JurisdictionModal):
const [selectedJurisdiction, setSelectedJurisdiction] = useState(''); 🤔
in that case there is no need to the type casting 🤔

Reply:
Hi @heorhi-deriv,
The API types for the shortcode is string in the `trading_platform_available_accounts` 🫠.
![Screenshot 2024-09-03 at 9 05 39 AM](https://github.com/user-attachments/assets/fc2da364-0914-4bde-927c-8ea0a462006b)

Reply:
Since this is the type which is carried from useQuery > useAvailableMT5Accounts > JurisdictionScreen > JurisdictionModal, I am not sure whether we can fix it.